pyccel/ast/numpytypes.py

```diff
@@ -176,6 +173,37 @@ class NumpyFloat64Type(NumpyNumericType):
     _primitive_type = PrimitiveFloatingPointType()
     _precision = 8
 
+    @property
+    def min_value(self):
+        """
+        Get the minimum representable value for this datatype.
+
+        This property returns the smallest value that can be represented by
+        this numeric datatype. It is useful for validation, boundary checks,
+        and ensuring that generated code respects the limits of the type.
+
+        Returns
+        -------
+        float64
+            The minimum value that can be represented.
+        """
+        return np.finfo(np.float64).min
```

Comment:
Why is this only needed for Float64?

To be honest I think I would have preferred if these `min_value`, `max_value` methods were left for a second PR. I would have been happier with the hard coded values and an issue to say that this needs to be done cleanly

Reply:
> Why is this only needed for Float64?

I.e. why is it not needed for Float32?

Reply:
It is needed for all numpy types i e
```python
a : 'float32[:]' ...
x = min( i+1 for i in a) 
```
this would require `min` and `max` values for `Float32`.

I need to add them in a separate PR and add tests for coverage.